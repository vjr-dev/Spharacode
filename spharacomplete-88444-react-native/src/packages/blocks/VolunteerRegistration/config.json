{
  "id": "VolunteerRegistration",
  "verion": "0.0.1",
  "status": "preAlpha",
  "path": "./packages/blocks/VolunteerRegistration",
  "assembler-adapters": [
    "VolunteerRegistrationAdapter"
  ],
  "mobile": {
    "merge-engine-navigation": {},
    "assembler-exports": [
      "VolunteerRegistration"
    ],
    "assembler-dependencies": [],
    "catalog-dependencies": [
      "core",
      "VolunteerRegistration"
    ],
    "assembler-devDependencies": [],
    "merge-engine-screens": {
      "VolunteerRegistration": {
        "file": "VolunteerRegistration.tsx",
        "extends": "VolunteerRegistrationController",
        "configURL": "",
        "validationRulesURL": "",
        "imports": [
          {
            "value": "VolunteerRegistrationController",
            "path": "./VolunteerRegistrationController"
          },
          {
            "value": "{Props}",
            "path": "./VolunteerRegistrationController"
          }
        ]
      }
    }
  }
  
}
