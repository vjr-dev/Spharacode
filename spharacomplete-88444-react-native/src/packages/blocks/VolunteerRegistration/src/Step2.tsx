
// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";

import {
    StyleSheet,
    Text,
    View,
    Image,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    TouchableOpacity,
    TextInput,
    Modal,
    ActivityIndicator,
    StatusBar

} from "react-native";

import VolunteerRegistrationController, { Props } from "./Step2Controller";
import { back1, back2, image_back, tick } from "./assets";
import { COLORS } from "../../../framework/src/Globals";
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;



export default class VolunteerRegistration extends VolunteerRegistrationController {
    constructor(props: Props) {
        super(props);
    }



    render() {
        return (
            <SafeAreaView style={styles.container1}>
                <View style={styles.child}>
                    <StatusBar
                        animated={true}
                        backgroundColor="#454545"
                        barStyle={"light-content"}
                        showHideTransition={"slide"}
                    // hidden={true}
                    />
                    <ImageBackground
                        source={back1}
                        style={styles.image}>
                        <ImageBackground
                            source={back2}
                            style={styles.image2}>


                            <View
                                style={styles.view1}
                            >
                                <TouchableOpacity
                                    testID="imageBackBtn"
                                    style={styles.button1} onPress={() => this.goback()}>
                                    <Image
                                        source={image_back}
                                        style={styles.backbutton}
                                    />
                                </TouchableOpacity>
                                {this.ison() ? <TouchableOpacity
                                    testID="varifycickBtn"
                                    style={styles.button1} onPress={() => this.Verifyclick()}>
                                    <Text style={{ color: COLORS.darkorange, marginRight: 20, fontWeight: "600" }}>VERIFY</Text>
                                </TouchableOpacity> : null}

                            </View>

                            <Text style={{ fontSize: 20, color: COLORS.white, marginTop: 50, width: windowWidth - 20, }}>
                                Step 2-Enter Confirmation Number</Text>
                            <Text style={{ fontSize: 15, color: COLORS.ultralightwhite, marginTop: 10, width: windowWidth - 20 }}>
                                Please enter confirmation number which you received from ypur local first-responder department.</Text>
                            <View style={{ flexDirection: "row", width: windowWidth - 20, alignSelf: "center", height: 50, alignItems: "center", marginTop: 50, backgroundColor: COLORS.Viewback, borderRadius: 50 }}>


                                <TextInput
                                testID="Otpcode"
                                    style={{ color: COLORS.ultralightwhite, marginLeft: 10, width: "100%" }}
                                    maxLength={4}
                                    value={this.state.Number}
                                    onChangeText={(value) => this.setState({ Number: value })}
                                    placeholder="4-Digit Code"
                                    placeholderTextColor={COLORS.ultralightwhite}
                                    keyboardType="number-pad"
                                    secureTextEntry={true}
                                />

                            </View>


                            <Modal
                                visible={this.state.Modal1}
                                // visible={true}
                                transparent={true}
                            >

                                <View
                                    style={{ width: windowWidth - 40, backgroundColor: "#fff", alignSelf: "center", marginTop: windowHeight / 4, borderRadius: 15 }}
                                >

                                    <View style={{ height: 50, alignItems: "center", justifyContent: 'center', marginTop: 20 }}
                                    >
                                        <Image
                                            source={tick}
                                            style={{ height: 30, width: 30, tintColor: '#f07233' }}
                                        />
                                    </View>
                                    <Text style={{ margin: 10, fontSize: 18, fontWeight: 'bold', alignSelf: "center" }}>Verification Done!</Text>
                                    <Text style={{ marginHorizontal: 10, fontSize: 15, alignSelf: "center", marginTop: 10, textAlign: "center", }}>Congratulation,You have become SPHARA verified volunteer</Text>
                                    <Text style={{ marginHorizontal: 10, fontSize: 18, alignSelf: "center", fontWeight: "bold", color: "#f07233", marginVertical: 20 }}></Text>
                                    <View style={{ height: 1, width: "100%", backgroundColor: "#808080", marginTop: 20 }} />
                                    <TouchableOpacity
                                    testID="modelContinue"
                                        onPress={() => this.modalclick()}
                                        style={{ height: 50, width: '100%', alignItems: "center", justifyContent: "center" }}>
                                        <Text style={{ fontWeight: '600', fontSize: 18 }}>CONTINUE</Text>

                                    </TouchableOpacity>
                                </View>
                            </Modal>

                            <Modal visible={this.state.Loader} transparent={true}>
                                <View style={{ height: windowHeight, width: windowWidth, alignItems: "center", justifyContent: "center" }}>
                                    <View
                                        style={{ height: 100, width: windowWidth - 20, alignSelf: "center", justifyContent: "center", alignItems: "center", }} >
                                        <ActivityIndicator
                                            animating={true}
                                            size={"large"}
                                            color="#f07233"
                                        />
                                    </View>
                                </View>
                            </Modal>

                        </ImageBackground>
                    </ImageBackground>
                </View>
            </SafeAreaView >

        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: windowHeight, width: windowWidth,
        backgroundColor: "#ffffffff"
    },
    image: {
        height: windowHeight, width: windowWidth,
    },
    image2: {
        height: windowHeight, width: windowWidth, alignItems: "center",
    },
    container1: { flex: 1, backgroundColor: "#454545" },
    child: { width: windowWidth, },
    backbutton: { height: 15, width: 15, tintColor: "#f07233", },
    view1: { height: 60, width: windowWidth, justifyContent: "space-between", flexDirection: "row", },
    button1: { marginLeft: 20, marginTop: 22 },

});
// Customizable Area End
