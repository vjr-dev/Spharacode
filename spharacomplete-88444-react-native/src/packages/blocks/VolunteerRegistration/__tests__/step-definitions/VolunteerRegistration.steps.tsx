//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'
import * as helpers from 'framework/src/Helpers'
import React from "react";
import VolunteerRegistration from "../../src/VolunteerRegistration"
import Step1 from "../../src/Step1"
import Step2 from "../../src/Step2"
import MessageEnum, { getName } from 'framework/src/Messages/MessageEnum';
import { Message } from 'framework/src/Message';
import { runEngine } from 'framework/src/RunEngine';



const navigation = require("react-navigation")
const CreateEventRes ={

}

const screenProps = {
    navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        addListener: jest.fn().mockImplementation((event: any, callback: any) => {
            callback();
          }),
    state: {
        params: {
          data: { name: 'Test Name' },
          userName: 'abc'
        }
      }
    },
    id: "VolunteerRegistration"
  }

  const screenPropsStep1 = {

    
    navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        addListener: jest.fn().mockImplementation((event: any, callback: any) => {
            callback();
          }),
    state: {
        params: {
          data: { name: 'Test Name1' },
          userName: 'abc1'
        }
      }
    },
    id: "Step1"
  }

  const screenProps3 = {
    navigation: {
      navigate: jest.fn(),
      goBack: jest.fn(),
      addListener: jest.fn().mockImplementation((event: any, callback: any) => {
          callback();
        }),
    },
    id: "Step2",
}



const feature = loadFeature('./__tests__/features/VolunteerRegistration-scenario.feature');


defineFeature(feature, (test) => {

 
    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');

        
    });
    test('User navigates to VolunteerRegistration', ({ given, when, then }) => {
        let VolunteerRegistrationBlockWrapper:ShallowWrapper;
        let instance:VolunteerRegistration ; 

        given('I am a User loading VolunteerRegistration', () => {
            //@ts-ignore
            VolunteerRegistrationBlockWrapper = shallow(<VolunteerRegistration {...screenProps}/>)
            expect(VolunteerRegistrationBlockWrapper).toBeTruthy()
            expect(VolunteerRegistrationBlockWrapper).toMatchSnapshot()     
        });

        when('I navigate to the VolunteerRegistration', () => {
             instance = VolunteerRegistrationBlockWrapper.instance() as VolunteerRegistration
      
        });


        then('VolunteerRegistration will load with out errors', () => {
      
          instance.setState({ VolunteerState: 1 })

           instance.setState({ Userdata: "" }, () => {})
          instance.setState({ switch1: "" })

            //@ts-ignore
            expect(VolunteerRegistrationBlockWrapper).toBeTruthy()
          
        });

      

        then('Handle back button press',() => {
            instance.goback()
            let buttonComponent = VolunteerRegistrationBlockWrapper.findWhere((node) => node.prop('testID') === 'goBackBtn');
            buttonComponent.simulate('press', instance.goback());

        })
      
      
        then("Handle the set button press", () => {
            const panictoggleComponent = VolunteerRegistrationBlockWrapper.findWhere(
              (node) => node.prop("testID") === "switch11Value"
            );
            panictoggleComponent.simulate("valueChange");
          });
      
          then('I can select the button with with out errors', () => {
           instance.goSignupScreen()
           instance.switch11(false)
           //instance. Updateapi()
            
           
        });
        then("VolunteerState condition true", () => {
          instance.setState({
            switch1:true,
            VolunteerState: 0,
          });
        });

        then("I can press ispublick screen with out errror", () => {
         
          const ispublickScreenComponent = VolunteerRegistrationBlockWrapper.findWhere(
            (node) => node.prop("testID") === "clickBtn"
          );
          ispublickScreenComponent.simulate("press");

          
        });
        then("I can press ispublick with out errror", () => {
          instance.setState({
            ispublick: true,
          });
        })    

        then("ispublick condition true", () => {
          instance.setState({
            ispublick:true,
          });
        });

        then("I can press modelOne screen with out errror", () => {
         
          const modelOneScreenComponent = VolunteerRegistrationBlockWrapper.findWhere(
            (node) => node.prop("testID") === "startClickID"
          );
          modelOneScreenComponent.simulate("press");
        });

        then("I can press modelOne with out errror", () => {
          instance.setState({
            Modal1: false,
          });
        }) 

        then("I can click modalbtn with out errror", () => {
          const modalbtnComponent = VolunteerRegistrationBlockWrapper.findWhere(
            (node) => node.prop("testID") === "startModelID"
          );
          modalbtnComponent.simulate("press");
        });
    

    
    
        
    })


   

    test('User navigates to Step1', ({ given, when, then }) => {
      let Step1BlockWrapper:ShallowWrapper;
      let instance:Step1 ; 

      given('I am a User loading Step1', () => {
          Step1BlockWrapper = shallow(<Step1 {...screenPropsStep1}/>)
            
      });

      

      when('I navigate to the Step1', async () => {
           instance = Step1BlockWrapper.instance() as Step1
           jest.setTimeout(30000);
           await instance.componentDidMount()
           instance.onclickModel()
           instance.apicaaal();
           instance.identificationResponse();
      },300000);


      then('Step1 will load with out errors', () => {
          expect(Step1BlockWrapper).toBeTruthy()
         

          const IndentificationListId = new Message(
            getName(MessageEnum.RestAPIResponceMessage)
          );
          IndentificationListId.addData(
            getName(MessageEnum.RestAPIResponceDataMessage),
            IndentificationListId.messageId
          );
          IndentificationListId.addData(
            getName(MessageEnum.RestAPIResponceSuccessMessage),
            JSON.parse(JSON.stringify(CreateEventRes))
          );
          instance.IndentificationListId = IndentificationListId.messageId;
          runEngine.sendMessage("Unit Test", IndentificationListId);


          const IndentificationId = new Message(
            getName(MessageEnum.RestAPIResponceMessage)
          );
          IndentificationId.addData(
            getName(MessageEnum.RestAPIResponceDataMessage),
            IndentificationId.messageId
          );
          IndentificationId.addData(
            getName(MessageEnum.RestAPIResponceSuccessMessage),
            JSON.stringify(CreateEventRes)
          );
          instance.IndentificationId = IndentificationId.messageId;
          runEngine.sendMessage("Unit Test", IndentificationId);
        
       
      });

      then('Handle back button press',() => {
       
        let buttonComponent = Step1BlockWrapper.findWhere((node) => node.prop('testID') === 'goBackID');
        buttonComponent.simulate('press');
      })
    
    
      then("Handle the set button press", () => {
        instance.setState({ modal2: false })
        instance.setState({ Loader: true });
        });

        then('I can select the button with with out errors', () => {
         

          let buttonComponent1 = Step1BlockWrapper.findWhere((node) => node.prop('testID') === 'modelOpenId');
          buttonComponent1.simulate('press')
          instance.onclickModel()

          let buttonComponent = Step1BlockWrapper.findWhere((node) => node.prop('testID') === 'model2ID');
          buttonComponent.simulate('press',instance.setState({ modal2: false }))


          let storageComponent = Step1BlockWrapper.findWhere((node) => node.prop('testID') === 'storageID');
          storageComponent.simulate('press')


      });

      then("LoaderState condition true", () => {
        instance.setState({
          Loader:false,
        });
      });

      then("I can press CameraID screen with out errror", () => {
       
        const CameraIDScreenComponent = Step1BlockWrapper.findWhere(
          (node) => node.prop("testID") === "CameraID"
        );
        CameraIDScreenComponent.simulate("press");

        const CameraIDScreenComponent1 = Step1BlockWrapper.findWhere(
          (node) => node.prop("testID") === "onClickID"
        );
        CameraIDScreenComponent1.simulate("press");
      }); 
  })


  test('User navigates to Step2', ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;
    let instance: Step2;

    given('I am a User loading Step2', () => {
        exampleBlockA = shallow(<Step2 {...screenProps3} />)
    });

    when('I navigate to the Step2', () => {
        instance = exampleBlockA.instance() as Step2
    });

    then('Step2 will load with out errors', () => {
        instance.ison()
        instance.Verifyclick()
        instance.ToggleVolunteerAPI()
      
        const ToggleVolunteerApiCallId = new Message(
          getName(MessageEnum.RestAPIResponceMessage)
        );
        ToggleVolunteerApiCallId.addData(
          getName(MessageEnum.RestAPIResponceDataMessage),
          ToggleVolunteerApiCallId.messageId
        );
        ToggleVolunteerApiCallId.addData(
          getName(MessageEnum.RestAPIResponceSuccessMessage),
          JSON.parse(JSON.stringify(CreateEventRes))
        );
        instance.ToggleVolunteerApiCallId = ToggleVolunteerApiCallId.messageId;
        runEngine.sendMessage("Unit Test", ToggleVolunteerApiCallId);


        const IndentificationId = new Message(
          getName(MessageEnum.RestAPIResponceMessage)
        );
        IndentificationId.addData(
          getName(MessageEnum.RestAPIResponceDataMessage),
          IndentificationId.messageId
        );
        IndentificationId.addData(
          getName(MessageEnum.RestAPIResponceSuccessMessage),
          JSON.stringify(CreateEventRes)
        );
        instance.IndentificationId = IndentificationId.messageId;
        runEngine.sendMessage("Unit Test", IndentificationId);

        expect(exampleBlockA).toBeTruthy()
        expect(exampleBlockA).toMatchSnapshot()
    });

    then('I can enter text with out errors', () => {
      
  });

 

    then('I can select the button with with out errors', () => {
        let buttonComponent = exampleBlockA.findWhere((node) => node.prop('testID') === 'imageBackBtn');
        buttonComponent.simulate('press')
        const fourDigitCode = exampleBlockA.findWhere(
          (node) => node.prop("testID") === "Otpcode")
          fourDigitCode.simulate("changeText", 4)

         expect(exampleBlockA).toMatchSnapshot();
         expect(instance.state.Number).toEqual(4)

         let buttonComponent1 = exampleBlockA.findWhere((node) => node.prop('testID') === 'modelContinue');
        buttonComponent1.simulate('press')
        
    });

});


}
)
