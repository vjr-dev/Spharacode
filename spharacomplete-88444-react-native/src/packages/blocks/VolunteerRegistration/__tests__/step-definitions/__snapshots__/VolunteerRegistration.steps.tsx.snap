// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolunteerRegistration User navigates to Step2 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "marginLeft": 20,
                "marginTop": 22,
              }
            }
            testID="imageBackBtn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          Step 2-Enter Confirmation Number
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 15,
              "marginTop": 10,
              "width": 730,
            }
          }
        >
          Please enter confirmation number which you received from ypur local first-responder department.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#515151",
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="number-pad"
            maxLength={4}
            onChangeText={[Function]}
            placeholder="4-Digit Code"
            placeholderTextColor="#a1a1a1"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "#a1a1a1",
                "marginLeft": 10,
                "width": "100%",
              }
            }
            testID="Otpcode"
            underlineColorAndroid="transparent"
            value={0}
          />
        </View>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={true}
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "marginTop": 333.5,
                "width": 710,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/VolunteerRegistration/assets/tick.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "tintColor": "#f07233",
                    "width": 30,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "margin": 10,
                }
              }
            >
              Verification Done!
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 15,
                  "marginHorizontal": 10,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Congratulation,You have become SPHARA verified volunteer
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#f07233",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                  "marginVertical": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#808080",
                  "height": 1,
                  "marginTop": 20,
                  "width": "100%",
                }
              }
            />
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="modelContinue"
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                CONTINUE
              </Text>
            </TouchableOpacity>
          </View>
        </Component>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;

exports[`VolunteerRegistration User navigates to Step2 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 60,
              "justifyContent": "space-between",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "marginLeft": 20,
                "marginTop": 22,
              }
            }
            testID="imageBackBtn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          Step 2-Enter Confirmation Number
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 15,
              "marginTop": 10,
              "width": 730,
            }
          }
        >
          Please enter confirmation number which you received from ypur local first-responder department.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#515151",
              "borderRadius": 50,
              "flexDirection": "row",
              "height": 50,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            keyboardType="number-pad"
            maxLength={4}
            onChangeText={[Function]}
            placeholder="4-Digit Code"
            placeholderTextColor="#a1a1a1"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "#a1a1a1",
                "marginLeft": 10,
                "width": "100%",
              }
            }
            testID="Otpcode"
            underlineColorAndroid="transparent"
            value={4}
          />
        </View>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={true}
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "marginTop": 333.5,
                "width": 710,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/VolunteerRegistration/assets/tick.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "tintColor": "#f07233",
                    "width": 30,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "margin": 10,
                }
              }
            >
              Verification Done!
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 15,
                  "marginHorizontal": 10,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Congratulation,You have become SPHARA verified volunteer
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#f07233",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                  "marginVertical": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#808080",
                  "height": 1,
                  "marginTop": 20,
                  "width": "100%",
                }
              }
            />
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="modelContinue"
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                CONTINUE
              </Text>
            </TouchableOpacity>
          </View>
        </Component>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;

exports[`VolunteerRegistration User navigates to VolunteerRegistration 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "height": 60,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "marginLeft": 20,
              }
            }
            testID="goBackBtn"
          >
            <Image
              source={
                Object {
                  "testUri": "../../../packages/blocks/VolunteerRegistration/assets/image_back.png",
                }
              }
              style={
                Object {
                  "height": 15,
                  "tintColor": "#f07233",
                  "width": 15,
                }
              }
            />
          </TouchableOpacity>
        </View>
        <Text
          style={
            Object {
              "color": "rgb(255,255,255)",
              "fontSize": 20,
              "marginTop": 50,
              "width": 730,
            }
          }
        >
          Your Volunteering Profile.
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 16,
              "marginTop": 10,
              "width": 730,
            }
          }
        >
          Set your volunteering profile type and ensure your
        </Text>
        <Text
          style={
            Object {
              "color": "#a1a1a1",
              "fontSize": 16,
              "marginTop": 5,
              "width": 730,
            }
          }
        >
          help to nearby victims by getting alerts.
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#404040",
              "flexDirection": "row",
              "height": 65,
              "marginTop": 40,
              "width": 750,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255,255,255)",
                "fontSize": 16,
                "marginLeft": 20,
                "width": "77%",
              }
            }
          >
            Intrested in volunteering?
          </Text>
          <Switch
            ios_backgroundColor="#000"
            onValueChange={[Function]}
            style={
              Object {
                "transform": Array [
                  Object {
                    "scaleX": 0.9,
                  },
                  Object {
                    "scaleY": 0.9,
                  },
                ],
              }
            }
            testID="switch11Value"
            thumbColor="rgb(255,255,255)"
            trackColor={
              Object {
                "false": "rgb(0,0,0)",
                "true": "rgba(255,138,73,1)",
              }
            }
            value={false}
          />
        </View>
        <Component
          hardwareAccelerated={false}
          testID="modalOpen"
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#fff",
                "borderRadius": 15,
                "marginTop": 333.5,
                "width": 710,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/VolunteerRegistration/assets/tick.png",
                  }
                }
                style={
                  Object {
                    "height": 30,
                    "tintColor": "#f07233",
                    "width": 30,
                  }
                }
              />
            </View>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "margin": 10,
                }
              }
            >
              Verification Done!
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 15,
                  "marginHorizontal": 10,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
            >
              Congratulation,You have become SPHARA verified volunteer
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#f07233",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginHorizontal": 10,
                  "marginVertical": 20,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#808080",
                  "height": 1,
                  "marginTop": 20,
                  "width": "100%",
                }
              }
            />
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
              testID="startModelID"
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "600",
                  }
                }
              >
                CONTINUE
              </Text>
            </TouchableOpacity>
          </View>
        </Component>
        <Component
          hardwareAccelerated={false}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 1334,
                "justifyContent": "center",
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": 100,
                  "justifyContent": "center",
                  "width": 730,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#f07233"
                hidesWhenStopped={true}
                size="large"
              />
            </View>
          </View>
        </Component>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
